thorlcr/graph/thgraphslave.hpp

```diff
@@ -70,15 +70,15 @@ class CEdgeProgress
             if (!hasStopped())
                 throw MakeActivityException(&owner, 0, "Starting without being stopped 1st");
             else
-                throw MakeActivityException(&owner, 0, "Started and stopped states both set");
+                count &= THORDATALINK_COUNT_MASK;
         }
         icount = 0;
         count = (count & THORDATALINK_COUNT_MASK) | THORDATALINK_STARTED;
     }
 
     inline void dataLinkStop()
     {
-        count = (count & THORDATALINK_COUNT_MASK) | THORDATALINK_STOPPED;
+        count |= THORDATALINK_STOPPED;
```

Comment:
I'm a bit concerned this changes the current semantics of hasStarted().
Currently (before this change), it would be false after stop, after this change, it will remain true.
So now means "has ever started".
There is a lot of code that is dependent on hasStarted(), and this may well cause that code to be recalled when it should not be.

Currently, it's impossible to track CQ activities start/stopped state - because of the single bit.
This change would allow it in theory to track the 'has it ever started' state, but that's not the same, and this change may break other things.

Reply:
@jakesmith Ok.  I understand the change is going to cause a problem.

I have examined the other ways of doing this. If we were to serialize the start and stop state separately, we could either have the master use the last recieved value or sum the value. If it were to use the last recieved value, it would always eventually end-up with start=false, stop=true.  If we can't really sum the values, as the NumStarts is likely to be serialized repeatedly for long-running activities.

The only thing I can think of is separating NumStart/NumStops from hasStarted().  Could we create a separate variable for hasStarted?